pkg/policy/l4.go

```diff
@@ -23,16 +23,15 @@ import (
 	"github.com/cilium/cilium/api/v1/models"
 	"github.com/cilium/cilium/pkg/labels"
 	"github.com/cilium/cilium/pkg/policy/api"
-	log "github.com/sirupsen/logrus"
 )
 
-const (
-	// WildcardEndpointSelector is a special hash value for the wildcard endpoint, i.e., applies to all
-	WildcardEndpointSelector = iota
+var (
+	// WildcardEndpointSelector is a selector that matches on all endpoints
+	WildcardEndpointSelector = api.EndpointSelector{}
 )
 
 // L7DataMap contains a map of L7 rules per endpoint where key is a hash of EndpointSelector
-type L7DataMap map[uint64]api.L7Rules
+type L7DataMap map[api.EndpointSelector]api.L7Rules
```

Comment:
This doesn't work. EndpointSelector is indeed comparable, but comparison will be based on the pointer fields. This doesn't de-duplicate the endpoint selectors. Therefore, you need to walk the whole map in any case, because you might have duplicates. So I'd suggest just making this a slice, and adding the EndpointSelector into each item in the slice. That will also look better in the `cilium endpoint get` output.

Reply:
The code is intentional as-is right now. De-duplication is not required. The append is required to account for the Wildcard instance in case multiple rules append to the wildcard.

I have no objections to beautify this later but it's not a strict requirement given the time constraint on having this make the 0.12 release.